src/integration-test/tests-intra-team.ts

```diff
@@ -0,0 +1,390 @@
+      // Store the deployment in the client
+      client.lastDeployment = {
+        ...deployment!.attributes,
+        id: deployment!.id,
+      };
+    });
+
+    it('should be able to fetch the raw deployment webpage', async function () {
+      const client = await clientFactory();
+      this.timeout(1000 * 30);
+      const url = client.lastDeployment!.url + '/index.html';
+      const response = await client.fetch(url);
+      expect(response.status).to.eq(200);
+    });
+
+    it('should be able to fetch deployment\'s screenshot', async function () {
+      const client = await clientFactory();
+      this.timeout(1000 * 60);
+      const response = await client.fetch(client.lastDeployment!.screenshot);
+      expect(response.status).to.eq(200);
+    });
+
+    it('should be able to fetch project\'s activity', async function () {
+      const client = await clientFactory();
+      this.timeout(1000 * 10);
+      const activities = await client.getProjectActivity()
+        .then(x => x.getEntities());
+      expect(activities).to.exist;
+      expect(activities).to.have.length(1);
+      expect(activities[0].attributes['activity-type']).to.equal('deployment');
+      expect(activities[0].attributes.deployment.status).to.equal('success');
+      expect(Number(activities[0].attributes.project.id)).to.equal(await client.lastProject!.id);
+      expect(activities[0].attributes.project.name).to.equal(projectName);
+      expect(activities[0].attributes.commit).to.exist;
+      expect(activities[0].attributes.branch.name).to.equal('master');
+    });
+  });
+
+  describe('configuring notifications', () => {
+
+    function testNotificationConfiguration(
+      configuration: NotificationConfiguration,
+      response: JsonApiEntity,
+    ) {
+      const id = Number(response.id);
+      expect(Number.isNaN(id)).to.be.false;
+      const attributes = response.attributes;
+      // Ensure that one or the other is defined
+      expect(!!(configuration.projectId || configuration.teamId)).to.be.true;
+      if (configuration.teamId) {
+        expect(Number(attributes['team-id'])).to.eq(configuration.teamId);
+      }
+      if (configuration.projectId) {
+        expect(Number(attributes['project-id'])).to.eq(configuration.projectId);
+      }
+      switch (configuration.type) {
+        case 'flowdock':
+          expect(attributes['flow-token']).to.eq(configuration.flowToken);
+          break;
+        case 'hipchat':
+          expect(attributes['hipchat-room-id']).to.eq(configuration.hipchatRoomId);
+          expect(attributes['hipchat-auth-token']).to.eq(configuration.hipchatAuthToken);
+          break;
+        case 'slack':
+          expect(attributes['slack-webhook-url']).to.eq(configuration.slackWebhookUrl);
+          break;
+      }
+      return id;
+    }
+
+    it('should be able to configure notifications', async function () {
+      const client = await clientFactory();
+      if (notificationConfigurations) {
```

Comment:
Should this be outside the `it(...)` statement, i.e. the tests are skipped if configurations don't exist? Now it'll give a false sense of security even if nothing is tested.